### DISCUSSION TITLE
Better language switcher

### DISCUSSION THREAD

**Discussion by JohnUW42** [jpanther/congo#462] [Ideas]
Hello!

In my opinion, the language switcher needs to be redesigned — it should be better positioned, next to the other menu items, because currently it looks kind of awkward right next to site's name / logo.

Language switcher also needs to be a drop-down menu — imagine that there not just 2 languages to choose from, but 10!

Maybe you could take the idea [from this](https://dev.to/feministclickback/how-to-add-a-language-switcher-to-your-hugo-site-2oh6) accessible css-only language switcher for Hugo.

**Screenshot 1 - current language switcher**

![Screenshot 2023-02-02 at 06 48 09](https://user-images.githubusercontent.com/87112040/216235483-91b1069b-9273-48ae-83db-f5519abb32f0.png)

**Screenshot 2 - current language switcher**

![Screenshot 2023-02-02 at 06 50 19](https://user-images.githubusercontent.com/87112040/216235487-21e6e4f5-5f27-4c6d-a2cf-57fb64a7d3e2.png)

Thank you!

**Comment by JohnUW42** [comment]
_Blist_ has a good language switcher:

https://github.com/apvarun/blist-hugo-theme

demo @ https://blist.vercel.app/

Though it does seem to be focused on accessibility.

**Comment by verlanmar** [comment]
Here's [the source](https://github.com/moodlebox/moodlebox.net/blob/a1858d9643da00ba62398d452454f95dc32ef542/layouts/partials/langswitcher.html) of the Hugo drop-down language switcher as seen on [moodlebox.net](https://moodlebox.net/).

**Comment by jpanther** [comment]
The language switcher is definitely on my radar for an overhaul. I was waiting until I had the foundations in place in the theme to be able to reposition links in menus. You would have seen in the last round of releases that the appearance switcher and search are now portable links and so my intention is to handle the language switcher in the same way.

That said, I haven't had time to implement this yet. The type of design I like is similar to what was raised in #290 and so I think I'll aim for that style of link, but with the option to place it anywhere. The _blist_ style is also nice and that would be in keeping with the design language of Congo however I haven't really given much consideration to CSS drop-down menus. That might be a future extension for the menus overall too.

**Comment by plbeaudoin** [comment]
If you are look for more inspiration:  https://www.droppath.app

I'm rebuilding the site to huge/congo.  The current version uses gulp+tailwindcss.

**Comment by AntoineSoetewey** [comment]
Hello,

Since v2.6.1, for websites with two languages, the fact that the current language is hidden and only the other one is displayed makes it hard for people to understand that it's actually a language switcher. For example, my website is in French and English, and when the English version is displayed, only the French flag is displayed next to the title:

<img width="1060" alt="Screenshot 2023-06-04 at 09 24 46" src="https://github.com/jpanther/congo/assets/17910063/0fa4ad01-2811-4389-8645-54ff37febc20">

I asked a few people, and most of them don't understand that they can click on the flag to display the French version of the website. Most of them simply thought that I wanted to print a French flag to my name.

With versions prior v2.6.0, all languages were displayed next to the title of the site. I believe it is indeed best to print all languages except the one which is displayed (as it is the case since v.2.6.0), but now that it may be the case that only one language is printed next to the title, I believe it would be best to position it in the menu as suggested by @plbeaudoin, @verlanmar and @JohnUW42.

What do you think @jpanther ?

As usual, thanks for the great work you put in this theme!

**Comment by jpanther** [comment]
I've created issue https://github.com/jpanther/congo/issues/583 to work on development of this feature. Please see my update there and provide additional feedback so that we can work towards a solution. I'll close this thread as I'd prefer to keep all the ongoing discussion in one place.
